app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,53 @@
+class MiqCockpitWsWorker::Authenticator
+  def authenticate_for_host(token, host)
+    user_obj = user_for_token(token)
+    return {} unless user_obj
+
+    # is it a container deployment
+    creds = find_container_node_creds(user_obj, host)
+
+    # Is it a VM
+    creds = find_vm_creds(user_obj, host) if creds.nil?
+
+    {
+      :valid  => true,
+      :known  => !creds.nil?,
+      :key    => !creds.nil? && creds.auth_key,
+      :userid => !creds.nil? && creds.userid,
+    }
+  end
+
+  private
+
+  # TODO: Do we need more user based permissions checks?
+  def find_container_node_creds(user_obj, host_or_ip)
+    raise "Looking up container nodes requires a valid user" unless user_obj
+    cond = "container_deployment_nodes.name = ? OR container_deployment_nodes.address = ?"
+    deployment = ContainerDeployment.joins(:container_deployment_nodes)
+                                    .find_by(cond, host_or_ip, host_or_ip)
```

Comment:
@petervo @jrafanie At a minimum, we should be doing this as a `.where`, because you can still do prepared statements from that:

```ruby
ContainerDeployment.joins(:container_deployment_nodes)
                   .where(cond, host_or_ip, host_or_ip).first
```

But if you want to do the condition as arel (not sure if there is an advantage here or not), you could probably do something like:

```ruby
cdn_table = ContainerDeployNode.arel_table
cond = cdn_table[:name].eq(host_or_ip).or(cdn_table[:address].eq host_or_ip)
deployment = ContainerDeployment.joins(:container_deployment_nodes).where(cond).first
```